semgrep_output_v1.atd

```diff
@@ -165,7 +165,25 @@ type error_severity
   | Info <json name="info">
 ]
 
-(* TODO? PRO can actually have "flavors" (see Engine_type.ml) *)
+(* What each flavor means:
+   - OSS = ran with OSS
+   - PRO = ran with PRO, but we didn't infer that OSS couldn't have found this
+   - PRO_SPECIFIC = ran with PRO and requires non-taint PRO features 
+   - PRO_SPECIFIC_INTRAFILE_TAINT = ran with PRO and requires intrafile taint
+   - PRO_SPECIFIC_INTERFILE_TAINT = ran with PRO and requires interfile taint
+   
+   OSS and PRO could have clearer names, but for backwards compatibility
+   we're leaving them as is *)
+type engine_flavor
+     <ocaml attr="deriving show">
+     <python decorator="dataclass(frozen=True)"> = [
+  | OSS
+  | PRO
+  | PRO_SPECIFIC
+  | PRO_SPECIFIC_INTRAFILE_TAINT
```

Comment:
Could this be a separate data type? `PRO_SPECIFIC of pro_specific_kind` ? I'm not great at naming but `PRO_ONLY` is shorter.

Reply:
Perhaps it could even be `OSS | PRO of pro_kind` ?

Reply:
But I don't think `PRO_SPECIFIC of something` breaks backwards compatibility, and having a flat structure here doesn't feel right. How many constructors we'll end up having if we later want to track more features? What if a finding involves multiple features and we want to record them all?